nixos/modules/system/boot/networkd.nix

```diff
@@ -1749,6 +1791,11 @@ in
     {
       systemd.network.units = mapAttrs' (n: v: nameValuePair "${n}.link" (linkToUnit n v)) cfg.links;
       environment.etc = unitFiles;
+
+      systemd.network.wait-online.extraArgs =
+        [ "--timeout=${toString cfg.wait-online.timeout}" ]
+        ++ optional cfg.wait-online.anyInterface "--any"
+        ++ map (i: "--ignore=${escapeShellArg i}") cfg.wait-online.ignoredInterfaces;
```

Comment:
`escapeShellArg` isn't valid for systemd Exec directives. this escaping should be done later, during interpolation into the directive.

Reply:
Indeed